### ISSUE TITLE
filter() output contains empty tables with CSV source

### ISSUE THREAD

**Issue by alespour** [influxdata/flux#2239]
When csv.from(...) is data source and filter condition is not met, result contains empty table for each such table. Such behaviour is not consistent with influxdb bucket data source and causes next operations in pipeline to fail since they expect tables containing data that match filter condition.

Tested with "flux repl" command. Flux version v0.55.1.

```
import "csv"

csvData = "#group,false,false,false,false,true,true,true
#datatype,string,long,dateTime:RFC3339,double,string,string,string
#default,_result,,,,,,
,result,table,_time,_value,_field,g4,_measurement
,,0,2019-11-01T00:00:00.209157349Z,-1.0,num,x,m

#group,false,false,false,false,true,true,true
#datatype,string,long,dateTime:RFC3339,string,string,string,string
#default,_result,,,,,,
,result,table,_time,_value,_field,g4,_measurement
,,1,2019-11-01T00:03:24.610806637Z,hithere,str,x,m
"

csv.from(csv: csvData) |> range(start: -1y) |> filter(fn: (r) => r.g4 == "no such value")

Result: _result
Table: keys: [_start, _stop, _field, g4, _measurement]
                   _start:time                      _stop:time           _field:string               g4:string     _measurement:string                      _time:time                  _value:float  
------------------------------  ------------------------------  ----------------------  ----------------------  ----------------------  ------------------------------  ----------------------------  
Table: keys: [_start, _stop, _field, g4, _measurement]
                   _start:time                      _stop:time           _field:string               g4:string     _measurement:string                      _time:time           _value:string  
------------------------------  ------------------------------  ----------------------  ----------------------  ----------------------  ------------------------------  ----------------------  
```

**Comment by jsternberg** [comment]
This is a duplicate of the filter inconsistency issue: #45.

**Comment by jsternberg** [comment]
This is fixed now by #2278.
